src/event.rs

```diff
@@ -421,3 +445,20 @@ macro_rules! event_subtype {
         }
     }
 }
+
+extern "C" fn event_handler_trampoline(event: *mut ffi::GdkEvent, ptr: glib_ffi::gpointer) {
+    let mut event = unsafe { from_glib_none(event) };
+    if ptr != ptr::null_mut() {
+        let raw: *mut Box<dyn Fn(&mut Event) + Send +'static> = ptr as _;
+        let f: &(dyn Fn(&mut Event) + Send +'static) = unsafe { &**raw };
+        f(&mut event)
```

Comment:
Should the event be mutable? I don't know much about GDK events, can they be changed in a meaningful way and does that even work with the bindings like this?

Reply:
I did this to match up with `gtk::main_do_event`. No idea really, though otherwise one must clone the event to call that. Maybe `main_do_event` shouldn't be mutable; would have to check GTK.

Reply:
Should be investigated then :)